### ISSUE TITLE
sporadic errors 

### ISSUE THREAD

**Issue by phlpn** [JuliaDB/ODBC.jl#67]
```
con = ODBC.connect("test")
res = query("select 1",con)
```
gives me random errors such as:
````
invalid UTF-8 sequence
while loading In[24], in expression starting on line 2

 in convert at utf8.jl:162
 in ODBCMetadata at  ..... \.julia\v0.3\ODBC\src\backend.jl:70
```
or
``` 
[ODBC] 42000: [Microsoft][SQL Server Native Client 10.0][SQL Server]Incorrect syntax near '}'.
[ODBC] 42000: [Microsoft][SQL Server Native Client 10.0][SQL Server]'...' is an invalid name because it contains a NULL character or an invalid unicode character.
[ODBC] 42000: [Microsoft][SQL Server Native Client 10.0][SQL Server]Unclosed quotation mark after the character string '...'.
[ODBC]: SQLExecDirect failed; Return Code: SQL_DEFAULT_ERROR
while loading In[36], in expression starting on line 2
```
Sometimes it gives the expected result. Any ideas?
I am running julia v0.3.1.

**Comment by aviks** [comment]
I'm seeing similar errors trying to query a MS SQL server from windows. It never works, but the error message changes. (tables and fields anonymised, but are all ascii)

```julia
julia> sql"SELECT TOP 1000 [gggg]
             ,[iiiii]
             ,[dddd]
         FROM [Xp].[dbo].[ssss]"
ERROR: invalid UTF-8 sequence

julia> sql"SELECT TOP 1000 [gggg]
             ,[iiiii]
             ,[dddd]
         FROM [Xp].[dbo].[ssss]"
ERROR: invalid UTF-8 sequence

julia> sql"SELECT TOP 1000 [ggg]
             ,[iiii]
             ,[dddd]
         FROM [Xp].[dbo].[ssss]"
[ODBC] 42000: [Microsoft][SQL Server Native Client 11.0][SQL Server]Incorrect syntax near 'P'.
ERROR: [ODBC]: SQLExecDirect failed; Return Code: SQL_DEFAULT_ERROR
```

**Comment by tristanmarkwell** [comment]
I'm having the same issues.  I'm on 64-bit Windows 7, 64-bit Julia, and my ODBC connection to the SQL Server database was set up using a 64-bit DSN.  The connection tests OK, and I get correct results from trivial queries such as "SELECT 1" and "SELECT GETDATE()" but as soon as I reference a table in the database I get "invalid UTF-8 sequence in convert at utf8.jl:163".  If I run the query repeatedly it seems to switch arbitrarily between this error and an error reference a syntax issue near a character not in my query (for example, [ODBC] 42000: [Microsoft][ODBC SQL Server Driver][SQL Server]Incorrect syntax near '>'.).

**Comment by haavardhvarnes** [comment]
I'm also having the same error as tristanmarkwell and with same configuration. The easy queries do work but I'm not getting any real data.

Any ideas?

**Comment by phlpn** [comment]
https://github.com/quinnj/ODBC.jl/pull/71 fixed it for me

**Comment by tristanmarkwell** [comment]
I tried to implement #71 (unfortunately due to my lack of understanding of how to use a specific branch of a package I just replaced the text of the 5 files in the ODBC folder with Stefan's text).  However, I'm now getting the opposite error: **invalid ASCII sequence in convert at ascii.jl:101**.  I don't get the syntax errors any longer.  Nothing in my query uses symbols not on the keyboard.  I hope it's not part of the issue but I'm running this through Juno since I'm struggling with the Julia-using-a-network-drive-to-store-packages issue with the REPL.

**Comment by Ivo-Balbaert** [comment]
I can confirm that removing the utf16(...) from ODBCQueryExecute in backend.jl line 44 makes it work.
I am on Windows 10, 64 bit., Julia 0.4.1

(I did not have this problem when I wrote "Getting started with Julia Programming", with Julia v 0.3)

**Comment by StefanKarpinski** [comment]
See https://groups.google.com/forum/#!topic/julia-users/BOeyuSlzCXE where much of this situation is hashed out.

**Comment by quinnj** [comment]
should be resolved by the overhaul 7ae7ebfac60ed3ffb1ccefd552993a3073f8b77b; please re-open if you're still seeing issues

**Comment by ax1ine** [comment]
I'm struggling with the same issue on Linux (ODBC.jl v. 1.0.4). Firebird ODBC driver accepts only UTF-8 strings but seems like ODBC.jl provides UTF-16. Is there any workaround presently?
